scripts/zones/Windurst_Waters_[S]/npcs/Mindala-Andola_CC.lua

```diff
@@ -1,29 +1,101 @@
 -----------------------------------
--- Area: Windurst Waters (S)
---   NPC: Mindala-Andola, C.C.
--- Type: Sigil
--- @zone 94
--- !pos -31.869 -6.009 226.793
---
--- Auto-Script: Requires Verification (Verified by Brawndo)
+-- Area: Windurst Waters [S]
+--  NPC: Andola-wtfever
+-- Type: Sigil NPC
 -----------------------------------
 package.loaded["scripts/zones/Windurst_Waters_[S]/TextIDs"] = nil;
 -----------------------------------
+require("scripts/globals/status");
+require("scripts/globals/campaign");
+require("scripts/zones/Windurst_Waters_[S]/TextIDs");
+-----------------------------------
 
 function onTrade(player,npc,trade)
 end;
 
 function onTrigger(player,npc)
-    player:startEvent(13, npc);
+    local notes = player:getCurrency("allied_notes");
+    local freelances = 99; -- Faking it for now
+    local unknown = 12; -- Faking it for now
+    local medalRank = getMedalRank(player);
+    local bonusEffects = 0; -- 1 = regen, 2 = refresh, 4 = meal duration, 8 = exp loss reduction, 15 = all
+    local timeStamp = 0; -- getSigilTimeStamp(player);
+
+    -- todo if Throne control is active
+
+    -- if ( medal_rank > 25 and nation controls Throne_Room_S ) then
+        -- medal_rank = 32;
+        -- this decides if allied ring is in the Allied Notes item list.
+    -- end
+
+    if (medal_rank == 0) then
+        player:startEvent(14);
+    else
+        player:startEvent(13, 0, notes, freelances, unknown, medalRank, bonusEffects, timeStamp, 0);
+    end
+
 end;
 
 function onEventUpdate(player,csid,option)
-    -- printf("CSID: %u",csid);
-    -- printf("RESULT: %u",option);
+    local itemid = 0;
+    local canEquip = 2; -- Faking it for now.
+    -- 0 = Wrong job, 1 = wrong level, 2 = Everything is in order, 3 or greater = menu exits...
+    if (csid == 13 and option >= 2 and option <= 2050) then
+        itemid = getWindurstNotesItem(option);
+        player:updateEvent(0, 0, 0, 0, 0, 0, 0, canEquip); -- canEquip(player,itemid));  <- works for sanction NPC, wtf?
+    end
 end;
 
 function onEventFinish(player,csid,option)
-    -- printf("CSID: %u",csid);
-    -- printf("RESULT: %u",option);
-end;
+    local medalRank = getMedalRank(player);
+    if (csid == 13) then
+        -- Note: the event itself already verifies the player has enough AN, so no check needed here.
```

Comment:
I'm semi wrong about that comment. Because of ppl injecting packets. If you'd like to handle that go ahead if not I'll do it later.